reana_db/models.py

```diff
@@ -147,6 +145,12 @@ class User(Base, Timestamp, QuotaBase):
     username = Column(String(length=255))
     tokens = relationship("UserToken", backref="user_", lazy="dynamic")
     workflows = relationship("Workflow", backref="user_", lazy="dynamic")
+    shared_workflows = relationship(
+        "Workflow",
+        secondary="__reana.user_workflow",
+        backref="shared_users",
+        lazy="dynamic",
+    )
```

Comment:
While looking into many-to-many relationships in SQLAlchemy, I have stumbled upon the warning in this section:  https://docs.sqlalchemy.org/en/13/orm/basic_relationships.html#association-object

It might be a good idea to make this relationship `shared_workflows` read-only (but I do not know whether this affects other parts of the workflow sharing code, so let me know if this is not feasible).

Reply:
Also, we could maybe change the name to `workflows_shared_with_me` or something similar? What do you think?
The current name is ambiguous, as it's not clear whether these are the workflows shared _with_ me or _by_ me

Reply:
I think that `viewonly=True` is not needed anymore now that you have removed `user_workflows` , but let's keep it like this for now